### PR TITLE
[Enhancement] Remove Hardcoded References to Master

### PR DESCRIPTION
**Pull request by MDK5**
### Motivation and Context
Since https://github.com/ls1intum/Artemis/pull/2844, Artemis supports any default branch. Orion currently has several hardcoded references to master that will not work with this change.

### Description
Instead of always pushing/pulling from/to master, Orion will now push/pull from the remote tracking branch of the current branch. If no such branch can be found, an error is show to the user. If multiple remote branches are present, a warning is shown to inform the user the changes might not have pushed to the main branch and therefore might not get graded.

### Steps for Testing
1.  Install the release as described in [the readme](https://github.com/ls1intum/Orion/blob/master/README.md#testing-of-pull-requests)
2.  Open a programming exercise as a student (make sure to use one where you have access to the git server)
3.  Test pushing (submitting) and pulling (automatically on startup) with various branch configurations:
    - Normal configuration with 1 branch
    - Create a local branch without remote branch (should give an error)
    - Create a different remote branch (should give a warning)
    - Switch to a different branch with remote tracking branch (should work, but give warning)
4. Also test an existing exercise to make sure no regression happened
5. (Optional) Create/Locate an exercise with the default branch not named master and try it. Theoretically the code does not mention "master" anymore, so it should not behave any different

### CODE REVIEW

File: src/main/resources/i18n/OrionBundle.properties [hunk 1]
```diff
@@ -22,13 +22,17 @@ orion.dialog.brokenlink.question=Do you want to link this exercise to Artemis ag
 orion.dialog.submissiondeletion.title=Submission will be deleted
 orion.dialog.submissiondeletion.label=Your currently open submission will be deleted when loading a new submission. Proceed?
 orion.vcs.updating=Updating your exercise files...
+orion.vcs.multipleremotes=Multiple remote branches detected. Your changes might not affect your grade if pushed to the wrong branch.
```
Comment (MDK5) [inline src/main/resources/i18n/OrionBundle.properties:25]:
```suggestion
orion.vcs.multipleremotes=Multiple remote branches detected. Your changes might got pushed to a branch that will not be assessed.
```

File: src/main/java/de/tum/www1/orion/vcs/OrionGitAdapter.kt [hunk 2]
```diff
@@ -271,32 +278,49 @@ object OrionGitAdapter {
     }
 
     /**
-     * This is equivalent to executing git reset --soft origin/master, which will make the local branch exactly the same
+     * This is equivalent to executing git reset --soft origin/branch, which will make the local branch exactly the same
```
Comment (MDK5) [inline src/main/java/de/tum/www1/orion/vcs/OrionGitAdapter.kt:281]:
```suggestion
     * This is equivalent to executing git reset --soft, which will make the local branch exactly the same
```

File: src/main/resources/i18n/OrionBundle_de.properties [hunk 3]
```diff
@@ -22,13 +22,17 @@ orion.dialog.brokenlink.question=Soll sie wieder mit Artemis verlinkt werden?
 orion.dialog.submissiondeletion.title=Abgabe wird gelöscht
 orion.dialog.submissiondeletion.label=Die momentane Abgabe wird durch das Herunterladen einer neuen gelöscht. Fortfahren?
 orion.vcs.updating=Aufgabendateien aktualisieren...
+orion.vcs.multipleremotes=Mehrere remote branches gefunden. Deine Änderungen wirken sich möglicherweise nicht auf deine Note aus, wenn sie in den falschen Branch gepusht werden.
```
Comment (MDK5) [inline src/main/resources/i18n/OrionBundle_de.properties:25]:
```suggestion
orion.vcs.multipleremotes=Mehrere remote branches gefunden. Deine Änderungen wurden möglicherweise auf einen Branch gepusht, der nicht bewertet wird.
```

File: src/main/resources/i18n/OrionBundle.properties [hunk 4]
```diff
@@ -22,13 +22,17 @@ orion.dialog.brokenlink.question=Do you want to link this exercise to Artemis ag
...
 orion.exercise.submissiondeletionfailed=Deletion of submission failed. Retry.
 orion.exercise.submissiondownloadfailed=Downloading of submission failed due to heap size. Download manually from the web client or increase IntelliJ heap size.
 orion.exercise.submissiondownloading=Downloading Submission...
 orion.warning.accessforbidden=Access Forbidden
 orion.warning.accessforbidden.message=Orion does not support opening pages outside of Artemis!
 orion.warning.accessforbidden.backtoprevious=You will be routed back to the previous page.
-orion.error.vcs.pushfailed=Error! Could not push to master
+orion.error.vcs.pushfailed.branch=Error! Could not determine remote branch to push to. Try pushing manually.
+orion.error.vcs.pushfailed.unknown=Error! Could not push to origin. Try pushing manually.
+orion.error.vcs.pullfailed.branch=Could not determine remote branch to reset to. Local repository might be out of sync. Try pulling manually.
```
Comment (MDK5) [inline src/main/resources/i18n/OrionBundle.properties:34]:
```suggestion
orion.error.vcs.pullfailed.branch=Could not determine remote branch to reset to. Local repository might be out of sync. Try to pull/update manually.
```

**Review by MDK5 — COMMENTED** [review]
From review

**Review by FlorianGlombik — APPROVED** [review]
Tested on ts2 for an exercise that used main as the default branch.
The warning was also displayed properly when changing the branch.

The code and translations look fine to me!

**Review by JohannesStoehr — APPROVED** [review]
Tested locally by connecting to ts2 using the main branch
Code looks good as well!

**Pull request merged**